app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -295,6 +352,30 @@ class AgendaRepositoryImpl(
         cancelAllNotifications()
     }
 
+    override suspend fun sendPersistedNotifications() {
+        // Adds ten seconds because scheduler does not schedule alarms even 1 second old
+        val timePlusTenSeconds = dateTimeConversion
+            .localDateTimeToZonedEpochMilli(LocalDateTime.now().plusSeconds(10))
+        val persistedNotifs = db.getPersistedNotifDao().getAllPersistedNotifs().first()
+
+        persistedNotifs.forEach {
+            scheduler.schedule(
+                agendaId = it.agendaId,
+                time = timePlusTenSeconds
+            )
+        }
+    }
+
+    override suspend fun upsertPersistedNotification(agendaId: String) {
+        db.getPersistedNotifDao().upsertPersistedNotif(
+            PersistedNotifEntity(agendaId = agendaId)
+        )
+    }
+
+    override suspend fun deleteAllPersistedNotifs() {
+        db.getPersistedNotifDao().deleteAllPersistedNotifs()
+    }
```

Comment:
I wonder why you need to save these notifications persistently? If you want to reschedule them after a device reboot, couldn't you just add a Room query to retrieve all future agenda items and use their times to schedule the notifications?